drivers/ieee802154/ieee802154_rf2xx_iface.c

```diff
@@ -0,0 +1,312 @@
+/* ieee802154_rf2xx_iface.c - ATMEL RF2XX IEEE 802.15.4 Interface */
+
+/*
+ * Copyright (c) 2019 Gerson Fernando Budke
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_MODULE_NAME ieee802154_rf2xx_iface
+#define LOG_LEVEL CONFIG_IEEE802154_DRIVER_LOG_LEVEL
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <errno.h>
+#include <assert.h>
+
+#include <device.h>
+#include <drivers/spi.h>
+#include <drivers/gpio.h>
+
+#include "ieee802154_rf2xx.h"
+#include "ieee802154_rf2xx_regs.h"
+#include "ieee802154_rf2xx_iface.h"
+
+void rf2xx_iface_phy_rst(struct device *dev)
+{
+	const struct rf2xx_config *conf = dev->config->config_info;
+	struct rf2xx_context *ctx = dev->driver_data;
+
+	/* Ensure control lines have correct levels. */
+	gpio_pin_write(ctx->reset_gpio, conf->reset.pin, 1);
+	gpio_pin_write(ctx->slptr_gpio, conf->slptr.pin, 0);
+
+	/* Wait typical time of timer TR1. */
+	k_busy_wait(330);
+
+	gpio_pin_write(ctx->reset_gpio, conf->reset.pin, 0);
+	k_busy_wait(10);
+	gpio_pin_write(ctx->reset_gpio, conf->reset.pin, 1);
+}
+void rf2xx_iface_phy_tx_start(struct device *dev)
+{
+	const struct rf2xx_config *conf = dev->config->config_info;
+	struct rf2xx_context *ctx = dev->driver_data;
+
+	/* Start TX transmission at rise edge */
+	gpio_pin_write(ctx->slptr_gpio, conf->slptr.pin, 1);
+	/* 16.125[μs] delay to detect signal */
+	k_busy_wait(20);
+	/* restore initial pin state */
+	gpio_pin_write(ctx->slptr_gpio, conf->slptr.pin, 0);
+}
+
+u8_t rf2xx_iface_reg_read(struct device *dev,
+			  u8_t addr)
+{
+	struct rf2xx_context *ctx = dev->driver_data;
+	u8_t status;
+	u8_t regval = 0;
+
+	addr |= RF2XX_RF_CMD_REG_R;
+
+	const struct spi_buf tx_buf = {
+		.buf = &addr,
+		.len = 1
+	};
+	const struct spi_buf_set tx = {
+		.buffers = &tx_buf,
+		.count = 1
+	};
+	const struct spi_buf rx_buf[2] = {
+		{
+			.buf = &status,
+			.len = 1
+		},
+		{
+			.buf = &regval,
+			.len = 1
+		},
+	};
+	const struct spi_buf_set rx = {
+		.buffers = rx_buf,
+		.count = 2
+	};
+
+	if (spi_transceive(ctx->spi, &ctx->spi_cfg, &tx, &rx) != 0) {
+		LOG_ERR("Failed to exec rf2xx_reg_read CMD at address %d",
+			addr);
+	}
+
+	LOG_DBG("Read Address: %02X, PhyStatus: %02X, RegVal: %02X",
+		(addr & ~(RF2XX_RF_CMD_REG_R)), status, regval);
+
+	return regval;
+}
+
+void rf2xx_iface_reg_write(struct device *dev,
```

Comment:
looks like reg_read and reg_write could use one function, easily factorizing the spi buffer setup. 
the common function would take a boolean to tell whether it's writing or reading, etc..

Reply:
Yes, is possible to refactor. I would like to understand the big advantage on do that. This way the method have one specific purpose and is very clear to read. Usually I try to avoid this type of construction and because of that is important to me understand this change. Could you please help me to see the benefit, please?

Reply:
It reduces the ROM usage basically. See for instance ieee802154_cc2520.c:z_cc2520_access()

Other than that, it's a taste issue. I am fine if you want to keep it as it is.

Reply:
I'll keep for now.